### PR TITLE
[develop]: Add GitHub Actions to check that Technical Docs are up-to-date

### LINKED ISSUE #1151: Check that Tech Docs are up-to-date
## Description
Currently, we have Technical Documentation, but there is no way to ensure that it stays up-to-date as scripts are added or removed. 

## Solution
Add a GitHub action that checks for up-to-date Tech Docs and fails when they have not been updated. 

## Requirements**
A test must fail if the technical documentation does not reflect the PR contents. 

## Acceptance Criteria (Definition of Done)
All PRs are check for up-to-date technical documentation and receive a failing test when they are out-of-date.

### PR DESCRIPTION
**Pull request by gspetro-NOAA**
## DESCRIPTION OF CHANGES: 
This PR:
- Adds a GitHub Actions workflow & script to check whether the Technical Documentation is up-to-date
- Turns on the `-W` flag so that documentation build warnings register as error
- Updates the requirements file so that `--remove-old` flag can be used to check Tech Docs
- Updates the Contributor's Guide w/information on Technical Documentation and troubleshooting guidelines
- Fixes a few broken links. 

### Type of change
- [X] New feature (non-breaking change which adds functionality)

## TESTS CONDUCTED: 
Ran fundamental tests on Hera & Hercules. All passed. 
I also tested the GitHub Actions functionality by adding or deleting test files and checking that the GitHub Actions behavior (pass/fail) corresponded to the expected behavior. 

- [ ] derecho.intel
- [ ] gaea.intel
- [ ] hera.gnu
- [x] hera.intel
- [x] hercules.intel
- [ ] jet.intel
- [ ] orion.intel
- [ ] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [ ] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## DEPENDENCIES:
N/A

## DOCUMENTATION:
Documentation was added to the Contributor's Guide to explain technical documentation standards. 

## ISSUE: 
Closes Issue #1151 

## CHECKLIST
- [x] My code follows the style guidelines in the Contributor's Guide
- [x] I have performed a self-review of my own code using the Code Reviewer's Guide
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain). N/A
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published N/A

### CODE REVIEW

File: .github/workflows/doc_tests.yaml
Comment (MichaelLueken) [inline .github/workflows/doc_tests.yaml]: Before this work get's merged, these two lines should be removed, since the feature/doc-actions branch will no longer require manual submission of the GitHub Action.

**Review by MichaelLueken — APPROVED** [review]
@gspetro-NOAA -

The changes look good to me!  I would suggest removing the `feature/doc-actions` branch from the `workflow_dispatch` in `.github/workflows/doc_tests.yaml`, since this feature branch won't require manual submission of the GHA once this PR is merged.

**Review by rickgrubin-noaa — APPROVED** [review]
LGTM -- approved pending change (removal of `github action`) asked for by @MichaelLueken.

File: .github/workflows/doc_tests.yaml
Comment (gspetro-NOAA) [reply .github/workflows/doc_tests.yaml]: Done!

Comment (MichaelLueken) [reply .github/workflows/doc_tests.yaml]: Thanks!

**Pull request merged**
